src/logservice/logminer/ob_log_miner_logger.cpp

```diff
@@ -50,25 +52,40 @@ int LogMinerLogger::log_progress(int64_t record_num, int64_t current_ts, int64_t
   int ret = OB_SUCCESS;
   double percentage = double(current_ts - begin_ts) / double(end_ts - begin_ts);
   double progress = 0;
+  struct winsize w;
+  int pb_width = 0;
   int lpad = 0;
   int rpad = 0;
   int64_t pos = 0;
   const ObString nls_format;
   char time_buf[128] = {0};
+  char pb_buf[MAX_SCREEN_WIDTH] = {0}; 
   // current_ts may exceed end_ts
   if (percentage > 1) {
     percentage = 1;
   }
   progress = percentage * 100;
-  lpad = (int)(percentage * PB_WIDTH);
-  rpad = PB_WIDTH - lpad;
+
+  if (!ioctl(STDOUT_FILENO, TIOCGWINSZ, &w)) { // get the width of terminal window
+    w.ws_col = (w.ws_col < MAX_SCREEN_WIDTH) ? w.ws_col : MAX_SCREEN_WIDTH;
```

Comment:
set variable `ret` if failed

Reply:
I think setting it to `MAX_SCREEN_WIDTH` when it fails might not be appropriate, the value is too large for a normal monitor.

Reply:
I think it doesn't matter if the `ioctl` fails, which means `pb_width` keeps `0`. We only cancel to print the progress bar at this time and it does not affect the return value of `LogMinerLogger::log_progress`.

Reply:
> i was mistaken, but it's better to print a warning log when it fails. `LOG_WARN_RET`

It makes sense.